packages/amplify-cli/src/domain/amplify-telemetry/VersionManager.ts

```diff
@@ -0,0 +1,23 @@
+import semver from 'semver';
+
+const APIVersionToPayloadVersion = new Map<string, Array<string>>([['v1.0', ['1.0.0']]]);
```

Comment:
What is the design here? a given CLI version just reports 1 version of the payload what we change over time. So the CLI does not have to support multiple versions.

On the other hand it seems that the `getLatestApiVersion()` is doing the version calculation on every call which seems to be suboptimal as this code is on the hot path if telemetry is enabled, if you decide to keep the multiple versions here, can we cache it between invocations?

Reply:
I want the API version to be mapped to payload version. This way it's explicit if there are any major changes we would have to make sure the API works with the new payload. No we are not caching it. It's kind of excessive to cache something that's been called twice in a lifetime